code/modules/mob/living/simple_animal/hostile/hostile.dm

```diff
@@ -245,6 +249,9 @@
 	if(isturf(the_target) || !the_target) // bail out on invalids
 		return FALSE
 
+	if(QDELETED(the_target) || QDELETED(src))
+		return FALSE
```

Comment:
```suggestion
	if(isturf(the_target) || QDELETED(the_target) || QDELETED(src)) // bail out on invalids
		return FALSE
```

QDELETED will do effectively the same thing as `!` in this instance, so let's just do it right and condense all the checks